src/figuresClasses.ts

```diff
@@ -1,11 +1,110 @@
-export interface Figure {}
+function roundTo(n: number, decimals: number): number {
+  const factor = Math.pow(10, decimals);
 
-export class Triangle implements Figure {}
+  return Math.floor(n * factor) / factor;
+}
+
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be positive numbers');
```

Comment:
The error message should specify which side is invalid. Consider using a more detailed message like `"Side 'a' must be a positive number"`.